openwisp_monitoring/device/schema.py

```diff
@@ -28,14 +28,43 @@
                     "type": "array",
                     "items": {"type": "number", "minItems": 3, "maxItems": 3},
                 },
+                "cpus": {"type": "integer"},
                 "memory": {
                     "id": "memory",
                     "type": "object",
                     "properties": {
                         "total": {"type": "integer"},
                         "free": {"type": "integer"},
                         "buffered": {"type": "integer"},
-                        "cache": {"type": "integer"},
+                        "cached": {"type": "integer"},
```

Comment:
here: define the required keys. Ensure that the key which are not required are accessed using `memory.get('<key>')` (which does not raise `KeyError`) rather than `memory['<key>']`